internal/safety/safety.go

```diff
@@ -37,12 +38,13 @@ func recoverFunc(fn func() error, withPanic bool) func() error {
 	return func() (err error) {
 		defer func() {
 			if r := recover(); r != nil {
-				log.Warnf("recovered:\t%+v", r)
+				stack := string(debug.Stack())
+				log.Warn("recovered: %+v\nstacktrace:\n%s", r, stack)
 				switch x := r.(type) {
 				case runtime.Error:
 					err = errors.ErrRuntimeError(err, x)
 					if withPanic {
-						log.Error(err, info.Get())
+						log.Errorf("recovered but this thread going to panic because panic reason is runtimer.Error\nerror: %v\ninfo:\n%s\nstacktrace:\n%s", err, info.Get().String(), stack)
```

Comment:
```suggestion
						log.Errorf("recovered but this thread going to panic: the reason is runtimer.Error\nerror: %v\ninfo:\n%s\nstacktrace:\n%s", err, info.Get().String(), stack)

```

Reply:
I thought that info.Get() already contains stack traces. is the `stack` different from that?

Reply:
I thought this StackTrace was the same, but our trace uses runtime.Caller, and runtime/debug.Stack uses buffered runtime.Stack and gets more information, so this is more informative for StackTrace.

Reply:
but runtime.Stack returns too many information, I think we need both information for debug and we should use debug.Stack only panic case.

Reply:
ok